### ISSUE TITLE
Cloudron LDAP integration

### ISSUE THREAD

**Issue by nebulade** [janeczku/calibre-web#2652]
I am currently trying to integrate calibre-web with the Cloudron LDAP server and hit some issues. This issue is to clarify if my understanding here is correct and to validate if it ever would be possible to integrate.

First, generally LDAP queries are working fine, so the connection and the admin binds are ok.

Basically from https://github.com/janeczku/calibre-web/wiki/LDAP-Login and debugging both server and the app, it seems that essentially in order to import users from LDAP groups, first the app queries all objects from `LDAP Distinguished Name` together with applying `LDAP Group Object Filter` substiting the `%s` with `LDAP Group Name`. On those results it would extract uids based on `LDAP Group Members Field` and then fetches those objects in return by querying again all objects from `LDAP Distinguished Name` with filter `LDAP User Object Filter`. Finally adding those results as users in the database.

Is this observation correct?

For Cloudron to give some background, our LDAP server implementation, as used by basically all other apps, would provide all users with access to the application on one user DN and then also all groups with access to the app on one group DN. The latter is optional though. So I am wondering if calibre-web can also be configured in a way to not rely on a specific group from the groups DN but import all users from a specific user DN?

**Comment by jdaviescoates** [comment]
> So I am wondering if calibre-web can also be configured in a way to not rely on a specific group from the groups DN but import all users from a specific user DN?

:crossed_fingers: :pray:

**Comment by OzzieIsaacs** [comment]
> Is this observation correct?

Yes

I'm not good at this LDAP stuff, to judge if this is possible and how I need a device tree (is it called like that?). Like in this file:(https://github.com/OzzieIsaacs/calibre-web-test/blob/main/test/helper_ldap.py -> config variable
And additionally an explanation for an total idiot what to search for and what to do.

**Comment by nebulade** [comment]
I spent some more time today debugging this and my initial observation was wrong.

Essentially first thing at https://github.com/janeczku/calibre-web/blob/master/cps/admin.py#L1604 is to list all users from a configured group and then work with those users. Such a group does not exist on Cloudron at least so for the moment there is no way to make this work well.

Further I tried to make it use a real group but there I can't get any further than this exception thrown via flask simpleldap: `ERROR {cps.admin:1605} not all arguments converted during string formatting` and after various trial and error attempts I guess I will give up as the docs for this at https://flask-simpleldap.readthedocs.io/en/latest/#flask_simpleldap.LDAP.get_group_members are also not really helpful
